Makefile

```diff
@@ -598,3 +602,14 @@ certs:
 .PHONY: certs-dryrun
 certs-dryrun:
 	cd pkg/config/tlscfg/testdata && ./gen-certs.sh -d
+
+.PHONY: repro-check
+repro-check:
+	# Check local reproducibility of generated executables.
+	$(MAKE) clean
+	$(MAKE) build-all-platforms
+	# Generate checksum for all executables under ./cmd
+	find cmd -type f -executable -exec sha256sum {} \; | sort -k2 > sha256sum.combined.txt
+	$(MAKE) clean
+	$(MAKE) build-all-platforms
+	sha256sum --strict --check ./sha256sum.combined.txt
```

Comment:
can we not use `shasum -a 256`? It should work on both Mac and Linux without additional dependencies

Reply:
`shasum` is included by `perl` package which is usually shipped with full Debian based distributions. I can't say that it will always be available in Linux as it's not part of coreutils. Particularly if this is run in a reduced size container which will likely have `perl-base` instead of `perl`.

Reply:
This would mostly just affect maintainers so it's a limited set of users.
Options:
1) Use `shasum` and require it to be installed by any edge-case Linux users. Presumably CI will function properly on [ubuntu-latest](https://github.com/jaegertracing/jaeger/blob/main/.github/workflows/ci-release.yml#L10) image, but it might not.
  - We could also leave it as-is in `package-deploy.sh` and only apply the chosen change to `Makefile`.
2) Require macOS users to do `brew install coreutils`.
3) Try to detect which commands exist and switch between the two.

Reply:
shasum works fine in our CI, I just tested: https://github.com/jaegertracing/jaeger/runs/8061728418?check_suite_focus=true#step:5:27